ncm-grub/src/main/perl/grub.pm

```diff
@@ -191,11 +193,25 @@ sub password
     my $val = $tree->{option} ? "--$tree->{option} " : "";
     $val .= $password;
 
-    $grub_fh->add_or_replace_lines(qr/^password\s+/,
-                                   qr/^password $val$/,
-                                   "password $val\n",
-                                   $self->main_section_offset($grub_fh),
-                                   );
+    if ( -d $GRUB2_DIR ) {
```

Comment:
Hmm, I'm not sure.  If you follow the method on the ubuntu wiki, and some methods I've read for el7, where you add "set superusers ...", "password_pbkdf2 ..." etc to a grub.d config, then sure you must run grub2-mkconfig.

But if you are making use of this standard grub.d, since it sources user.cfg itself, so then I don't think it's necessary. Do you see what I mean, or am I missing something?

# grub2-mkconfig >/tmp/grub.cfg.old
/usr/bin/grub2-editenv: error: cannot rename the file /boot/grub2/grubenv.new to /boot/grub2/grubenv: No such file or directory.
Generating grub configuration file ...
Found linux image: /boot/vmlinuz-3.10.0-514.6.1.el7.x86_64
Found initrd image: /boot/initramfs-3.10.0-514.6.1.el7.x86_64.img
Found linux image: /boot/vmlinuz-0-rescue-3a304c440aca4b8bbbd43fe3b4db7297
Found initrd image: /boot/initramfs-0-rescue-3a304c440aca4b8bbbd43fe3b4db7297.img
done
# mv user.cfg user.cfg.bak
# grub2-mkconfig >/tmp/grub.cfg.new
/usr/bin/grub2-editenv: error: cannot rename the file /boot/grub2/grubenv.new to /boot/grub2/grubenv: No such file or directory.
Generating grub configuration file ...
Found linux image: /boot/vmlinuz-3.10.0-514.6.1.el7.x86_64
Found initrd image: /boot/initramfs-3.10.0-514.6.1.el7.x86_64.img
Found linux image: /boot/vmlinuz-0-rescue-3a304c440aca4b8bbbd43fe3b4db7297
Found initrd image: /boot/initramfs-0-rescue-3a304c440aca4b8bbbd43fe3b4db7297.img
done
# diff -Naur /tmp/grub.cfg.old /tmp/grub.cfg.new
#